hashintel/hash
# Add boolean & number type support to entity editor

## 🌟 What is the purpose of this PR?

This PR adds functionality for viewing/editing value cells with data type of "Number" and "Boolean" at entity editor.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203150072758509/1203301695650120/f) _(internal)_
- [Design](https://app.asana.com/0/1203150072758509/1203301695650120/f) _(internal)_

## 🔍 What does this change?




- Add `No value` state to value cells
- Move `isValueEmpty` to a separate file
- Add draw & edit support for boolean & number values

## Next Steps

- Match the design of boolean selector 1-1 with design (this PR does not modify the default `Select` style we have in the project)

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Open entity editor with boolean/number types
1.  Confirm that editing/viewing boolean/number types work & look like expected

## 📹 Demo
 
### Boolean view
<img width="848" alt="Screenshot 2022-11-09 at 13 57 28" src="https://user-images.githubusercontent.com/39553853/200812594-c5bebc91-59b0-442f-b6be-ebb82fc540f3.png">

### Boolean editor
<img width="843" alt="Screenshot 2022-11-09 at 13 57 33" src="https://user-images.githubusercontent.com/39553853/200812600-c1cf020b-8c6f-4142-b1f2-013cc0b1f7ed.png">

### Number editor

<img width="844" alt="Screenshot 2022-11-09 at 13 57 39" src="https://user-images.githubusercontent.com/39553853/200812606-ce3690f3-5836-4448-bc84-60549c645c5b.png">
